### ISSUE TITLE
Failed with: quic-go doesn't build on Go 1.18 yet.

### ISSUE THREAD

**Issue by CompuRoot** [caddyserver/xcaddy#90]
`xcaddy`: from development branch
`go version go1.18 linux/amd64`

The same issue as it was in [4216](https://github.com/caddyserver/caddy/issues/4216)

`xcaddy` successfully compiled, but command `xcaddy list-modules` failed with error on following:

```
go/pkg/mod/github.com/lucas-clemente/quic-go@v0.23.0/internal/qtls/go118.go:5:13: cannot use "quic-go doesn't build on Go 1.18 yet." (untyped string constant) as int value in variable declaration
```

**Comment by francislavoie** [comment]
The latest stable version of Caddy was released before Go 1.18 was released, so it doesn't support it. We made fixes on the master branch to support Go 1.18, so if you absolutely need to build with 1.18, you should build from master for now. Otherwise, please downgrade to 1.17 in the meantime until Caddy's next stable release.

**Comment by tyler71** [comment]
If you build with docker and can't easily downgrade your go version, `caddy:2.4.1-builder` worked fine
